share/completions/zypper.fish

```diff
@@ -29,12 +31,50 @@ function __fish_zypper_use_repo
 end
 
 function __fish_zypper_print_repos
-    for repofile in /etc/zypp/repos.d/*.repo
-        string replace -f -r '\[(.+)\]' '$1' < $repofile
+    zypper -t lr | sed '1,/-+-/d' | cut -d '|' -f 2 | cut -d ' ' -f 2
```

Comment:
What does the `zypper` output look like here? I'd love it if we could do this with our `string` tool instead of invoking external things.

Reply:
I'm assuming this should offer "Base_System", "code", "fish-nightly" and such?

If so, `string replace` can be used to transform lines (with a regex-mode with `-r`/`--regex`), and with `-f`/`--filter` it'll only print lines that matched, so that you don't need an explicit `string match`.

Something like

```fish
zypper -t lr -E | string replace -rf '^[\d\s]+\| (\S+)\s*\| (\S+).*$' '$1\t$2'
```

should work. The trick is to make the regex tight enough to ignore the first couple of lines.

In this case, the bit that throws out the other stuff is that we expect the line to start with numbers and whitespace up to a pipe character and a space - `^[\d\s]+\| `. We then capture the next two fields, throwing away whitespace. We can use this to turn the line into effectively `$alias\t$name`, which means the name is used as a description.